profiles/tests/test_audit_log.py

```diff
@@ -276,17 +276,35 @@ def test_audit_log_create(cap_audit_log):
 """
 
 
-def test_actor_is_resolved_in_graphql_call(live_server, profile, cap_audit_log):
+def test_actor_is_resolved_in_graphql_call(
+    live_server, profile, service_client_id, cap_audit_log
+):
+    service = service_client_id.service
+    ServiceConnectionFactory(profile=profile, service=service)
     user = profile.user
-
-    do_graphql_call_as_user(live_server, user, query=MY_PROFILE_QUERY)
+    do_graphql_call_as_user(live_server, user, service=service, query=MY_PROFILE_QUERY)
     audit_logs = cap_audit_log.get_logs()
     assert len(audit_logs) == 1
     log_message = audit_logs[0]
     assert_common_fields(log_message, profile, "READ", actor_role="OWNER")
     assert log_message["audit_event"]["actor"]["user_id"] == str(user.uuid)
 
 
+def test_service_is_resolved_in_graphql_call(
+    live_server, profile, service_client_id, cap_audit_log
+):
+    user = profile.user
+    service = service_client_id.service
+    do_graphql_call_as_user(live_server, user, service=service, query=MY_PROFILE_QUERY)
```

Comment:
Hmm, just noticed that the calling User has no authority to request `myProfile` with the specified Service, as the Service isn't "implicitly connected" and there's no explicit service connection either. With the peculiarities of the audit logging system, the log line is produced before the authorization is checked and this test passes. But it would still be good to change the test so that it doesn't raise the "permission denied" error it does now. That way whenever the peculiarities are removed from the audit logging system, this test will still work without changes.

Reply:
True. I'll add a service connection there.